crates/net/eth-wire-types/src/broadcast.rs

```diff
@@ -654,6 +653,17 @@ pub struct ValidAnnouncementData {
 
 handle_mempool_data_map_impl!(ValidAnnouncementData,);
 
+impl From<PartiallyValidData<Eth68TxMetadata>> for ValidAnnouncementData {
+    /// Conversion from [`PartiallyValidData`] from an announcement. Note! [`PartiallyValidData`]
+    /// from an announcement, should have some [`EthVersion`]. Panics if [`PartiallyValidData`] has
+    /// version set to `None`.
+    fn from(data: PartiallyValidData<Eth68TxMetadata>) -> Self {
+        let PartiallyValidData { data, version } = data;
+        let version = version.expect("should have eth version for conversion");
+        Self { data, version }
+    }
+}
```

Comment:
Here the compiler was throwing a strange error with the `From` derivative and the `from_partially_valid_data` function. So I figured it made more sense to do a classic `From` impl, hence this new implementation that seems cleaner and straightforward.

Reply:
`From` shouldn't fail
> Note: This trait must not fail. The From trait is intended for perfect conversions. If the conversion can fail or is not perfect, use TryFrom.

Reply:
Yes I know but it was the exact same behaviour as old function

```rust
pub fn from_partially_valid_data(data: PartiallyValidData<Eth68TxMetadata>) -> Self {
        let PartiallyValidData { data, version } = data;

        let version = version.expect("should have eth version for conversion");

        Self { data, version }
    }
```

which had the same goal. But in fact I realized that we could simply leave everything as before and not use the derive From which is no longer compatible here because we don't need it with the `from_partially_valid_data` method which does the same job.

I just cleaned up normally it's good.